frontend/src/app/accountCreation/AccountCreationApp.tsx

```diff
@@ -63,10 +68,13 @@ const AccountCreationApp = () => {
       // Set the userAccountStatus state, triggering a rerender w/ the Router
       setUserAccountStatus(status);
     };
-    const token = getActivationTokenFromUrl();
-    getStatusAndActivate(token);
-    /* eslint-disable react-hooks/exhaustive-deps */
-  }, []);
+
+    if (initializeToken.current) {
+      initializeToken.current = false;
+      const token = getActivationTokenFromUrl();
+      getStatusAndActivate(token);
+    }
+  }, [location.pathname, navigate]);
```

Comment:
question: would we ever get an unwanted rerender adding these values to the dependency array? Notice that we're changing this useEffect from being a "run once on app load" to depending on these values in React Router changing.

Don't have the context for RR at all though, so totally could be missing something!

Reply:
a few things happened behind this code change. 
1- Even when the dependency array was empty this logic was triggering more than once during cypress testing. So to make sure this only triggers once an initialization flag has been added.
2- Given that the logic to activate the token is now guarded behind the flag I added the dependencies to the array as this is actually the way is supposed to happen. Pathname as primitive value and navigate as ref passed as a hook  are not triggering weird renders.